src/main/java/org/craftercms/studio/impl/v2/dal/DataSourceInitializerImpl.java

```diff
@@ -243,12 +248,14 @@ public void initDataSource()  throws DatabaseUpgradeUnsupportedVersionException,
                             logger.error("Integrity validator error after running upgrade DB scripts", e);
                         }
 
-                        if (dbVersion.startsWith(DB_VERSION_3_0_X)) {
+                        if (initialDbVersion.startsWith(DB_VERSION_3_0_X) ||
+                                initialDbVersion.equals(DB_VERSION_3_1_0) ||
+                                initialDbVersion.equals(DB_VERSION_3_1_0_1)) {
```

Comment:
Why is `DB_VERSION_3_1_0_1` needed? are you planning on adding a new field for each update to the db?

Reply:
no, just these two are affected besides 3.0

Reply:
Every time the DB schema is updated, even if the version has not released, the DB version is updated with a snapshot version. The latest one is 3.1.0.2.